inc/commonitilobject.class.php

```diff
@@ -794,12 +794,15 @@ protected function handleTemplateFields(array $input) {
       if (count($tt->mandatory)) {
          $mandatory_missing = [];
          $fieldsname        = $tt->getAllowedFieldsNames(true);
+         $is_first_update   = $this->fields['date_creation'] == $this->fields['date_mod'];
          foreach ($tt->mandatory as $key => $val) {
             if ((!$check_allowed_fields_for_template || in_array($key, $allowed_fields))
                 && (isset($input[$key])
                     && (empty($input[$key]) || ($input[$key] == 'NULL'))
-                    // Take only into account already set items : do not block old tickets
-                    && (!empty($this->fields[$key]))
+                    // Take only into account already set items : do not block
+                    // old tickets unless they haven't been modified yet, this
+                    // is useful for recurrent ticket with mandatory params.
```

Comment:
Is it possible to also check if we're in a recurrent case to limit?

Reply:
Do we have this information ? I was looking for a `ticketrecurrent_ids` in `glpi_tickets` but didn't find anything.

Reply:
I really do not know; I'll have to take a look in the code to see what I can find

Reply:
As far as I've seen, we have the `glpi_ticketrecurrents` table which contains a `tickettemplates_id` field; but I do not know if that can be used; or if we can have the information here (tehre are very few direct calls on TicketRecurent).

Reply:
To use `tickettemplates_id` from `glpi_ticketrecurrents` we need to do Ticket -> Template -> TicketRecurrent.
I don't think it's possible, the template used to create a ticket is not stored so we can't do Ticket -> Template.

Also since this information is never stored, tickets are actually checked against the default template on update.
This seems a bit weird, if you use a template to validate your ticket creation then you should use the same template to validate updates ?

Maybe a deeper issue here ? Should we add this link ?
@orthagh 
@cedric-anne

Reply:
Indeed, the `*templates_id` is never stored (issues is same for `Change`, `Problem` and `Ticket`), so, on update, the used template corresponds to the template that matches tickets configuration.

Template is defined on creation/update using these rules:
1) Use template defined in choosed category for given type, if configured.
2) Fallback to template configured in active profile, if configured.
3) Fallback to template configured in item entity, if configured.
4) Apply category + type from template matching profile/entity and retry step 1.

Use cases:
a) For templates defined by choosed category and item type, unless category configuration changed, the same template will be used on update.
b) For templates defined by entity configuration, unless ticket is transfered of entity configuration changed, the same template will be used on update.
c) But for templates defined by profile, template will depend on user that update the ticket.

On update, template are used to hide fields and prevent unsetting a mandatory field that was already set.
If we store the `templates_id` and reuse it on update, result could have problematic impacts on usecase `c`.
Indeed, template used on ticket creation can give access to more fields that the template that would be affected by the user profile. So we could unintendedly give access to more fields to a user than expected.


An alternative solution would be to store the `ticketrecurrents_id` of the ticket, in order to be able to pass it on `CommonITILObject::getITILTemplateToUse()` method, to limit impact to only recurrent tickets, but it may lead to same problems.